### DISCUSSION TITLE
Welcome to HQ Discussions!

### DISCUSSION THREAD

**Discussion by jackiekazil** [Open-Source-Software-Neighborhood-Watch/town-square#2] [Announcements]
## 👋 Welcome! Please introduce yourself!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jspeed-meyers** [comment]
I'm John Speed Meyers, but friends call me John Speed.

A public policy researcher by background, I've gotten interested in free and open source software over that past couple years. I spent a lot of my pandemic virtually meeting people around the world that are interested in open source software and security! I've worked on R&D related to [Python package security](https://www.iqt.org/auraborealis-how-we-found-20-vulnerable-python-packages/) and on how [software developers do or don't use security criteria](https://www.iqt.org/code-reuse-holy-grail-or-poisoned-chalice/) when choosing packages. I then had the good fortune of meeting several people in this group and here I am!

I currently help organize the bi-weekly meeting where we share projects we're working on and give feedback.

**Comment by tpike3** [comment]
Hi, I'm Tom

I am interested in understanding how the open source ecosystems work as problem solving network. Currently, some colleagues and I (to include John and Jackie)  are doing some research with the Python ecosystem and have some intriguing initial results.  We hope to then expand to CRAN, Julia and NPM. Ideally, this will help these ecosystems to identify their strengths and weaknesses and understand how they can function more efficiently and effectively, and be more resilient.  We also believe understanding the open source ecosystem can help us understand how to work together, more effectively. We will share the results and code in the Open Source Neighborhood Watch once we solidify our initial findings.

**Comment by jackiekazil** [comment]
👋 I am Jackie, and I feel like I am terrible at introductions. 🤣 

I am well integrated into the Python community, but have a passion for the whole open source ecosystem, because we are in this together.  I am leader in the global [Pyladies](https://pyladies.com/) community, and in the past, I have served as a board member of the [Python Software Foundation](https://www.python.org/psf/). I have a passion for the work that @tpike3 described above and computational social science as a whole.

**Comment by gps-iqt-labs** [comment]
👋 Hello World! I'm George and I work a lot with npm, among other package managers.

I find open source ecosystems and community-developed software projects fascinating. I've previously collaborated with @jspeed-meyers, @jackiekazil, and @tpike3, who are all incredibly talented and welcoming people, and I'm excited to see where this community goes next.

**@future-participants**, I hope you'll introduce yourselves too. 👇

**Comment by DemetersSon83** [comment]
Interesting article:

**Crowdsourcing the open-source security problem**

"We’ve known about the shaky foundation of open-source software security for years, but in the wake of the discovery of the Log4j vulnerability last year, we’re starting to see more action.

After the Biden administration gathered leaders from across enterprise tech and open-source software communities to discuss how to improve open-source security, new initiatives from The Linux Foundation, Google, Microsoft and npm have all targeted the need for better security hygiene within open-source software. On Tuesday npm’s parent company, GitHub, rolled out a few changes to its database of known vulnerabilities that could allow developers better access to information in their daily workflow and an opportunity to share helpful details with their colleagues.

    GitHub’s Advisory Database is now open to contributions from developers who might want to share detailed information about software vulnerabilities, such as the versions that are affected.
    The Microsoft-owned coding repository also published the “full contents” of that database in a new format that will be available under a very permissive license, [it said in a blog post](https://github.blog/2022-02-22-github-advisory-database-now-open-to-community-contributions/).
    “With community contributions, security researchers, academics, and enthusiasts will now be able to provide additional information and context to further the community’s understanding and awareness of security advisories,” GitHub’s Kate Catlin wrote in the post.

For more than 20 years, the [CVE](https://www.cve.org/) list has been the gold standard for official information on security vulnerabilities, but [that information goes through a deliberate process](https://www.cve.org/About/Process).

    Whenever a critical software vulnerability is discovered, it sets off a race between users of that software scrambling to defend themselves and hackers bent on exploiting the vulnerability.
    GitHub users closer to the action don’t have the authority that an organization like CVE has, but they often have tips and tricks that could help users determine if they are affected by the vulnerability and what steps to take next.
    An internal GitHub team [will still review the submissions](https://github.com/github/advisory-database) to guard against the expected craziness that always comes along with allowing the public to post things to the internet, and it will also use the [Open Source Vulnerabilities](https://osv.dev/) format that allows computers to read and process the listings.

While the new endeavor appears to be a good-faith effort by GitHub to help find ways to improve an enormous industry-wide problem, it’s hard to overlook that it also helps GitHub’s long-term vision to be the central tool in a software developer’s toolkit.

    Developers have lots of information resources at their disposal, but it’s not hard to imagine that information presented alongside their day-to-day coding tasks would decrease the incentive to go elsewhere.
    Still, given the degree to which enterprise tech relies on open-source software to run its operations, more sources of information on security vulnerabilities are always helpful.
    “By making it easier to contribute to and consume, we hope it will power even more experiences and will further help improve the security of all software,” Catlin wrote.

— Tom Krazit ([email](mailto:tkrazit@protocol.com) | [twitter](http://twitter.com/tomkrazit))"

https://www.protocol.com/newsletters/protocol-enterprise/github-open-source-security-singularity
